jedi/inference/references.py

```diff
@@ -140,10 +144,15 @@ def find_references(module_context, tree_name):
             search_name,
         )
 
+    gotos = set()
+    if not all_scopes:
+        for name in found_names_dct.values():
+            gotos |= set(name.goto())
```

Comment:
With the modification removed below, I think you won't need the `gotos` variable anymore. The filter below will probably be as simple as `.get_root_context() == module_context` (where module_context is the original param that was not modified by the loop variable).